app/auth/create.js

```diff
@@ -0,0 +1,101 @@
+import React, { useState, useEffect } from 'react'
+import { Text, TextInput, HelperText, Button } from 'react-native-paper'
+import { useRouter } from 'expo-router'
+import { useDispatch, useSelector } from 'react-redux'
+
+import AuthPage from '@components/AuthPage'
+import { createUser, selectIsLoggedIn } from '@store'
+
+const CreateAccount = () => {
+    const dispatch = useDispatch()
+    const router = useRouter()
+
+    const [fName, setFName] = useState('')
+    const [lName, setLName] = useState('')
+    const [email, setEmail] = useState('')
+    const [password, setPassword] = useState('')
+    const [passwordConfirm, setPasswordConfirm] = useState('')
+    const [badEmail, setBadEmail] = useState(false)
+    const [passwordsMatch, setPasswordsMatch] = useState(true)
+
+    useEffect(() => {
+        const re = /^[\w-\\.]+@crimson.ua.edu$/
+        if (email.length > 0 && !re.test(email)) {
+            setBadEmail(true)
+        } else {
+            setBadEmail(false)
+        }
+    }, [email])
+
+    useEffect(() => {
+        if (passwordConfirm.length > 0 && passwordConfirm == password) {
+            setPasswordsMatch(true)
+        } else {
+            setPasswordsMatch(false)
+        }
+    }, [isLoggedIn])
```

Comment:
Why is the dependency for this useEffect hook the isLoggedIn variable? That's not making sense to me. Shouldn't the dependencies be passwordConfirm and password, so that we perform a check when either of them changes?

Reply:
Shouldn't `password` also be a dependency? Otherwise someone could bypass the check by getting the `confirmPassword` equal to `password` and then changing `password` after the fact.